maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/RawXJC2Mojo.java

```diff
@@ -479,6 +481,7 @@ protected void doExecute() throws MojoExecutionException {
 			setupDirectories();
 			doExecute(options);
 			addIfExistsToEpisodeSchemaBindings();
+            deleteEpisodeIfOnlyFileOnEmptyContext();
```

Comment:
Wouldn't it be better to just not call `doExecute` instead of creating and then deleting the file?

Reply:
@alerosmile : there is some scenario (simple ones with no bindings / catalog) where files are just created the right way

Reply:
@laurentschoelens : You are right but I once read this:
https://www.eclipse.org/lists/m2e-dev/msg01178.html

maven-resources-plugin:copy-resources does nothing during onConfiguration.

Reply:
So you suggest that the runOnConfiguration should be set to false ?

Reply:
No, the source roots must be added during runOnConfiguration, but according to Igor's comment, no files should be created. I'm not sure but I think Igor is one of the main creator of m2e.

Reply:
In fact, it's working as expected with only runOnIncremental 😄

Reply:
> In fact, it's working as expected with only runOnIncremental 😄

Or not 😄